internal/translation/networkpeering/networkpeering.go

```diff
@@ -0,0 +1,147 @@
+package networkpeering
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/provider"
+)
+
+type PeerConnectionsService interface {
+	CreatePeer(ctx context.Context, projectID string, conn *NetworkPeer) (*NetworkPeer, error)
+	ListPeers(ctx context.Context, projectID string) ([]NetworkPeer, error)
+	DeletePeer(ctx context.Context, projectID, containerID string) error
+}
+
+type PeeringContainerService interface {
+	CreateContainer(ctx context.Context, projectID string, container *ProviderContainer) (*ProviderContainer, error)
+	GetContainer(ctx context.Context, projectID, containerID string) (*ProviderContainer, error)
+	ListContainers(ctx context.Context, projectID, providerName string) ([]ProviderContainer, error)
+	DeleteContainer(ctx context.Context, projectID, containerID string) error
+}
+
+type NetworkPeeringService interface {
+	PeerConnectionsService
+	PeeringContainerService
+}
+
+type networkPeeringService struct {
+	peeringAPI admin.NetworkPeeringApi
+}
+
+func NewNetworkPeeringService(peeringAPI admin.NetworkPeeringApi) NetworkPeeringService {
+	return &networkPeeringService{peeringAPI: peeringAPI}
+}
+
+func (np *networkPeeringService) CreatePeer(ctx context.Context, projectID string, conn *NetworkPeer) (*NetworkPeer, error) {
+	atlasConnRequest, err := toAtlasConnection(conn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert peer to Atlas: %w", err)
+	}
+	newAtlasConn, _, err := np.peeringAPI.CreatePeeringConnection(ctx, projectID, atlasConnRequest).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create network peer %v: %w", conn, err)
+	}
+	newConn, err := fromAtlasConnection(newAtlasConn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert peer from Atlas: %w", err)
+	}
+	return newConn, nil
+}
+
+func (np *networkPeeringService) ListPeers(ctx context.Context, projectID string) ([]NetworkPeer, error) {
+	var peersList []NetworkPeer
+	providers := []provider.ProviderName{provider.ProviderAWS, provider.ProviderAzure, provider.ProviderGCP}
+	for _, providerName := range providers {
+		peers, err := np.listPeersForProvider(ctx, projectID, providerName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list network peers for %s: %w", string(providerName), err)
+		}
+		peersList = append(peersList, peers...)
+	}
+	return peersList, nil
+}
+
+func (np *networkPeeringService) listPeersForProvider(ctx context.Context, projectID string, providerName provider.ProviderName) ([]NetworkPeer, error) {
+	results := []NetworkPeer{}
+	pageNum := 1
```

Comment:
it's the first time I see paging being used. Why do we need to do this here and not elsewhere?

Reply:
I think elsewhere it was a bug, because usually it will not be needed, but we have no guarantees some config might not pass one page.

Still this might not be needed now that peers are individually managed instead of being part of a list. I will need to revisit it.

Reply:
Ohhh! In that case please open a bug ticket to investigate List operations everywhere! We use lists in a couple of translation layer services already without paging.

Reply:
Please investigate if a List without paging yields **all** results. If so, paging is not necessary here.

Reply:
CLOUDP-285975